src/FSharp.Analyzers.Cli/Program.fs

```diff
@@ -160,21 +172,23 @@ let runProjectAux
                     | None -> false
             )
             |> Array.map (fun fileName ->
-                let fileContent = File.ReadAllText fileName
-                let sourceText = SourceText.ofString fileContent
+                async {
+                    let! fileContent = File.ReadAllTextAsync fileName |> Async.AwaitTask
+                    let sourceText = SourceText.ofString fileContent
+                    logger.LogDebug("Checking file {0}", fileName)
+
+                    // Since we did ParseAndCheckProject, we can be sure that the file is in the project.
+                    // See https://fsharp.github.io/fsharp-compiler-docs/fcs/project.html for more information.
+                    let! parseAndCheckResults = fcs.GetBackgroundCheckResultsForFileInProject(fileName, fsharpOptions)
```

Comment:
As the comment mentions, this helps with speedily getting the typecheck for that file since we already checked the project.